### ISSUE TITLE
Get errors when trying to create a commit

### ISSUE THREAD

**Issue by dylanirlbeck** [tpope/vim-fugitive#1472]
When I try and run `cc` in the fugitive buffer to create a commit with some staged files, I get a long list of the same error showing up. After quitting the buffer with `:q` and then returning to fugitive, the files do not show up as staged. This is the error I'm seeing:

```
(node:19565) Warning: The 'NO_COLOR' env is ignored due to the 'FORCE_COLOR' env being set.
(node:19565) Warning: The 'NO_COLOR' env is ignored due to the 'FORCE_COLOR' env being set.
(node:19565) Warning: The 'NO_COLOR' env is ignored due to the 'FORCE_COLOR' env being set.
... (on and on)
```

Has anyone seen this before and/or know a fix? Thanks in advance!

**Comment by tpope** [comment]
Do you use lint-staged? It's caused issues before, such as #1446.

**Comment by dylanirlbeck** [comment]
Yes I do! I'll check out #1446 and see if I can fix this.

**Comment by tpope** [comment]
This will likely need to be addressed upstream. The warning is mostly worthless: Environment variables don't have a single origin so this sort of thing is inevitable, and there's not a clear action to be taken when it does. But it's *completely* worthless in this case because they're both set to disable colors!

You can work around it by doing a `unset NO_COLOR` in your `pre-commit` hook.

**Comment by dylanirlbeck** [comment]
@tpope Weird, I'm still getting the error after setting this as my pre-commit hook

```es6
 "husky": {
    "hooks": {
      "pre-commit": "unset NO_COLOR && lint-staged"
    }
  },
```

Is there anything else I need to do?

**Comment by alxndr** [comment]
@dylanirlbeck
> Weird, I'm still getting the error after setting this as my pre-commit hook
> 
> ```js
>  "husky": {
>     "hooks": {
>       "pre-commit": "unset NO_COLOR && lint-staged"
>     }
>   },
> ```
> 
> Is there anything else I need to do?

Try `unset` as a separate statement: `unset NO_COLOR; lint-staged`

**Comment by tpope** [comment]
If that really is the correct way to unset it (`&&` shouldn't matter but I have otherwise no idea how husky works) it sounds like the issue is purely internal, bumping the "likely" to "definitely".

The warning doesn't activate when `FORCE_COLOR` is set to `0`, so something is setting it to the wrong value:

https://github.com/nodejs/node/blob/46062a0d2c105aa8344e1f6f11ccb447a223b78d/lib/internal/tty.js#L112-L128

**Comment by tpope** [comment]
Actually we haven't ruled out external causes. Make sure `:echo exists('$FORCE_COLOR')` returns `0`.

**Comment by alxndr** [comment]
Interesting — I've changed `pre-commit` to be  "NO_COLOR=0 lint-staged", and now I see a _much smaller_ wall of text, but still with a few color codes:

    husky > pre-commit (node v12.4.0)
    ^[[?25l  ^[[33m↓^[[39m Stashing changes... ^[[2m[skipped]^[[22m
        ^[[90m→ No partially staged files found...^[[39m
      ^[[33m❯^[[39m Running tasks...
        ^[[33m❯^[[39m Running tasks for *.{js,jsx,json,scss}
          ^[[33m⠙^[[39m prettier --write
            git add
        ^[[33m↓^[[39m Running tasks for *.js ^[[2m[skipped]^[[22m
        ^[[90m  → No staged files match *.js^[[39m
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m↓^[[39mStashing changes... ^[[2m[skipped]^[[22m
        ^[[90m→ No partially staged files found...^[[39m
      ^[[32m✔^[[39m Running tasks...
    ^[[?25h^[[?25h
    Press ENTER or type command to continue

...previously I had been seeing 3 screenfuls of output, perhaps due to the spinner "animation"? Here is the first pageful...

    husky > pre-commit (node v12.4.0)
    ^[[?25l  ^[[33m⠙^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m⠹^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m⠸^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m⠼^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m⠴^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m⠦^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[33m⠧^[[39m Stashing changes...
        Running tasks...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[32m✔^[[39m Stashing changes...
      ^[[33m❯^[[39m Running tasks...
        ^[[33m❯^[[39m Running tasks for *.{js,jsx,json,scss}
          ^[[33m⠙^[[39m prettier --write
            git add
        ^[[33m❯^[[39m Running tasks for *.js
          ^[[33m⠙^[[39m eslint --fix
            git add
        Updating stash...
        Restoring local changes...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[32m✔^[[39m Stashing changes..
    .
      ^[[33m❯^[[39m Running tasks...
        ^[[33m❯^[[39m Running tasks for *.{js,jsx,json,scss}
          ^[[33m⠹^[[39m prettier --write
            git add
        ^[[33m❯^[[39m Running tasks for *.js
          ^[[33m⠹^[[39m eslint --fix
            git add
        Updating stash...
        Restoring local changes...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[32m✔^[[39m Stashing changes..
    .
      ^[[33m❯^[[39m Running tasks...
        ^[[33m❯^[[39m Running tasks for *.{js,jsx,json,scss}
          ^[[33m⠸^[[39m prettier --write
            git add
        ^[[33m❯^[[39m Running tasks for *.js
          ^[[33m⠸^[[39m eslint --fix
            git add
        Updating stash...
        Restoring local changes...
    ^[[?25l^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[1A^[[2K^[[G  ^[[32m✔^[[39m Stashing changes..
    .
      ^[[33m❯^[[39m Running tasks...
        ^[[33m❯^[[39m Running tasks for *.{js,jsx,json,scss}
          ^[[33m⠼^[[39m prettier --write
            git add
    -- More --

**Comment by tpope** [comment]
I would think you would want `NO_COLOR=1`, not `0`.

**Comment by tpope** [comment]
Or I guess you are trying to that as an alternative to `unset`, but I would not expect that to be equivalent to `unset` either.

**Comment by ianchanning** [comment]
`unset NO_COLOR; lint-staged` worked for me to remove the warning.

**Comment by ianchanning** [comment]
However lint-staged did also fix this problem in v10.0.4: https://github.com/lint-staged/lint-staged/issues/781#issuecomment-579588138

I was on v9.5, but can confirm that v15 of lint-staged made the warning go away. Nice collaboration between @tpope and lint-staged :smile:
